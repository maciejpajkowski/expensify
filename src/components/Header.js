import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogOut } from '../actions/auth';

export const Header = ({ startLogOut }) => (
    <header>
      <h1>Expenses Calculator</h1>
      <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
      <NavLink to="/create" activeClassName="is-active">Create an expense</NavLink>
      <NavLink to="/help" activeClassName="is-active">Halp pls</NavLink>
      <button onClick={startLogOut}>Log out</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogOut: () => dispatch(startLogOut())
});

export default connect(undefined, mapDispatchToProps)(Header);