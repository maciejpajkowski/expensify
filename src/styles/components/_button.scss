.button {
    border: 0;
    background-color: $blue;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}