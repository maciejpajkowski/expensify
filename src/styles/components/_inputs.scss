.text-input {
    border: 1px solid #cacccd;
    height: 50px;
    padding: $s-size;
    font-size: $font-size-large;
    font-weight: 300;
}

.select {
    @extend .text-input;
    font-size: 1rem;
    padding: 0 $xs-size;
}

.textarea {
    @extend .text-input;
    height: 10rem;
}