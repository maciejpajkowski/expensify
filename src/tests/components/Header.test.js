import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { Header } from '../../components/Header';
import { shallow } from 'enzyme';

test('should render Header correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Header startLogOut={() => { }} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should call startLogOut on button click', () => {
  const startLogOut = jest.fn();
  const wrapper = shallow(<Header startLogOut={startLogOut} />);
  wrapper.find('button').simulate('click');
  expect(startLogOut).toHaveBeenCalled();
});