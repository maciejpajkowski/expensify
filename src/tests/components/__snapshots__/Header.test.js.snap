// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expenses Calculator
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create an expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    Halp pls
  </NavLink>
  <button
    onClick={[Function]}
  >
    Log out
  </button>
</header>
`;
